plugins/inputs/directory_monitor/directory_monitor.go

```diff
@@ -357,6 +389,10 @@ func (monitor *DirectoryMonitor) Init() error {
 		monitor.fileRegexesToIgnore = append(monitor.fileRegexesToIgnore, regex)
 	}
 
+	if err := choice.Check(monitor.ParseMethod, []string{"line-by-line", "at-once"}); err != nil {
+		return fmt.Errorf("config option parse_method: %w", err)
+	}
```

Comment:
I would set the default method here and make the error message a bit more clear. This avoids the requirement to set the default method in every test etc.
```suggestion
	if monitor.ParseMethod == "" {
		monitor.ParseMethod = defaultParseMethod
	}
	if err := choice.Check(monitor.ParseMethod, []string{"line-by-line", "at-once"}); err != nil {
		return fmt.Errorf("invalid config option parse_method: %w", err)
	}
```

Reply:
Yes, that makes sense indeed.

About the error message, this is what it would look like in my proposal:
> config option parse_method: unknown choice xxxxxx

Reply:
Yes as it captures cases where the user sets an empty value in the config. Furthermore, this will make the test for the default **automatic** in tests. With your current implementation, tests will not necessarily capture changes to the default if someone decides not to change the `defaultParseMethod` but changing the value directly in `init`.

Reply:
Okay, good point, will change it. Shouldn't the Init and tests also be covered to also have this for the other default values as well?

Reply:
On the other hand. Having an empty string is incorrect configuration to me. The docs clearly say what the possible values can be..